### ISSUE TITLE
Copy mode of publishDir follows symbolic links instead of copying them

### ISSUE THREAD

**Issue by zichner** [nextflow-io/nextflow#592]
When running the following process:

```
process test {
	publishDir "outputDir", mode: 'copy'

	output:
	file '*' into testOutput

	"""
	echo "TEST" > testFile.txt
	ln -s testFile.txt testFileLink.txt
	"""
}
```

both result files (`testFile.txt` and `testFileLink.txt`) end up as regular files in `outputDir`. However, the expectation would be that `testFileLink.txt` stays a symbolic link.

**Comment by pditommaso** [comment]
Wait, I was understanding the other way around. Copying the concrete file instead of the symlink is the expected behaviour. What's the value of copying the link instead of the target file? 

Also considering the above example the copied link will be broken.

**Comment by zichner** [comment]
I think there are multiple use cases where it makes sense to copy all output files as they are, i.e., if a file is a link it is copied as such instead of following the link.

Here is one example:
Some tools generate many output files in a very specific directory structure. In order to access certain files later more conveniently, I want to generate symbolic links in the top directory pointing to them. For example, I get the following directory structure:

```
dir1
  dir1.1
    dir1.1.1
    dir1.1.2
      file1
  dir1.2
dir2
  dir2.1
    file2
```

To access the files `file1` and `file2` easier, I create (relative) symbolic links in the top directory (pointing to `dir1/dir1.1/dir1.1.2/file1` and `dir2/dir2.1/file2`).

If I use `publishDir` with `mode: copy`, the two links in the top directory become entire copies of the files (both files are contained twice in the output, consuming twice the space). However, my expectation would be, that the two links stay relative symbolic links (i.e., using `cp -P` instead of `cp -L`).

Having this as a separate `publishDir mode` would be great.

PS: In the example of my initial post, the symbolic link would not be broken, since it is pointing to a relative rather than absolute location.

**Comment by pditommaso** [comment]
Yes, it makes sense. I tend to agree that a new  copy mode would be the better way to handle this. Something like `copyLinks`, `linkCopy`, `cplink`, ..  any suggestion?

**Comment by zichner** [comment]
Maybe something like `copyNoDeref` or `copyNoDereference` for "copying without de-referencing symbolic links" (in `man cp` this functionality is called `--no-dereference`)

**Comment by pditommaso** [comment]
This is good but a bit too formal, we tend to use short and more intuitive names.
